vumi/persist/tests/test_model.py

```diff
@@ -133,6 +133,43 @@ def reverse_from_3(self, migration_data):
         migration_data.set_value('text', migration_data.old_data['text'])
         return migration_data
 
+    def migrate_from_3(self, migration_data):
+        # Migrator assertions
+        assert self.data_version == 3
+        assert self.model_class is IndexRemovedVersionedModel
+        assert isinstance(self.manager, Manager)
+
+        # Data assertions
+        assert set(migration_data.old_data.keys()) == set(
+            ['$VERSION', 'c', 'text'])
+        assert migration_data.old_data['$VERSION'] == 3
+        assert migration_data.old_index == {"text_bin": ["hi"]}
```

Comment:
We can't make these assertions outside of the migration functions. Given that these migrations are written specifically for the tests and include any assertions at all, I don't think the more data-specific ones are entirely unreasonable.